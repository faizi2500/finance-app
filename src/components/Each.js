import React from 'react';
import PropTypes from 'prop-types';
import { FaArrowRight } from 'react-icons/fa';
import { NavLink } from 'react-router-dom';

const Each = ({
  name,
  price,
  index,
  company,
}) => {
  let boxStyle = 'oddBox';

  if (index % 2 === 0) {
    boxStyle = 'evenBox';
  }
  return (
    <div className={boxStyle}>
      <FaArrowRight className="arrow" />
      <NavLink
        className="navLink"
        to={{
          pathname: `/company${name}`,
          name,
        }}
      >
        <h4 className="companyName">{company}</h4>
      </NavLink>
      <div className="combine-details">
        <h3>{name}</h3>
        <h3>
          $
          {price}
        </h3>
      </div>
    </div>
  );
};

Each.propTypes = {
  name: PropTypes.string.isRequired,
  price: PropTypes.string.isRequired,
  index: PropTypes.number.isRequired,
  company: PropTypes.string.isRequired,
};

export default Each;
