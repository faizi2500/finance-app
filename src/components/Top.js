import React from 'react';
import { FaMicrophone, FaCog } from 'react-icons/fa';

const Top = () => (
  <div className="top-body">
    <div className="holder">
      <p className="top-heading">Gainers</p>
      <div className="icons">
        <FaMicrophone />
        <FaCog />
      </div>
    </div>
  </div>
);

export default Top;
