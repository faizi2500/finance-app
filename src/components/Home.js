import React from 'react';
import PropTypes from 'prop-types';
import Each from './Each';

const Home = ({ nasdaq }) => {
  const companies = nasdaq;
  companies.sort((a, b) => b.changesPercentage - a.changesPercentage);
  return (
    <>
      <div className="full-width">
        <h5 className="list-heading">Top Performers</h5>
      </div>
      <div className="container">
        {companies.map((item, index) => (
          // eslint-disable-next-line max-len
          <Each key={item.ticker} name={item.ticker} price={item.price} index={index} company={item.companyName} />
        ))}
      </div>
    </>
  );
};

Home.propTypes = {
  nasdaq: PropTypes.instanceOf(Object).isRequired,
};

export default Home;
