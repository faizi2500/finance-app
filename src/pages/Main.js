import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { listAPIcall } from '../redux/home/home';
import Top from '../components/Top';
import Home from '../components/Home';
import cover from '../assets/cover.jpg';

const Main = () => {
  const dispatch = useDispatch();
  useEffect(() => {
    dispatch(listAPIcall());
  }, []);
  const result = useSelector((state) => state.countriesReducer);
  return (
    <>
      <Top />
      <img className="cover-img" src={cover} alt="Grow" />
      <div>
        <Home nasdaq={result} />
      </div>
    </>
  );
};

export default Main;
