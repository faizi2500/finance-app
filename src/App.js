import React from 'react';
import { Switch, Route } from 'react-router';
import Main from './pages/Main';
import Company from './pages/Company';

const App = () => (
  <>
    <Switch>
      <Route path="/company:name">
        <Company />
      </Route>
      <Route path="/company">
        <Company />
      </Route>
      <Route exact path="/">
        <Main />
      </Route>
    </Switch>
  </>
);

export default App;
