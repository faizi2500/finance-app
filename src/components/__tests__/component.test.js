import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter as Route } from 'react-router-dom';
import store from '../../redux/configureStore';
import App from '../../App';
import Main from '../../pages/Main';
import Each from '../Each';
import Home from '../Home';
import Top from '../Top';

test('renders correctly', () => {
  const home = renderer
    .create(
      <Provider store={store}>
        <Route>
          <App />
        </Route>
      </Provider>,
    )
    .toJSON();
  expect(home).toMatchSnapshot();
});

test('Comapnies component renders correctly', () => {
  const tree = renderer
    .create(
      <Route>
        <Provider store={store}>
          <Main />
        </Provider>
      </Route>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test('Companies Data Passed', () => {
  const mockStore = [
    {
      changes: 1.97,
      changesPercentage: '39.32135',
      companyName: 'Grove, Inc.',
      price: '11.91',
      ticker: 'GRVI',
    },
    {
      changes: 2.47,
      changesPercentage: '"26.165258"',
      companyName: 'NRx Pharmaceuticals, Inc.',
      price: '6.98',
      ticker: 'NRXP',
    },
  ];
  const tree = renderer
    .create(
      <Route>
        <Provider store={store}>
          <Home nasdaq={mockStore} />
        </Provider>
      </Route>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test('Banner of home page displaying', () => {
  const tree = renderer
    .create(
      <Route>
        <Provider store={store}>
          <Top />
        </Provider>
      </Route>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test('each company data shows', () => {
  const i = 1;
  const tree = renderer
    .create(
      <Route>
        <Provider store={store}>
          <Each name="AAPL" price="450" index={i} company="APPLE INC" />
        </Provider>
      </Route>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
