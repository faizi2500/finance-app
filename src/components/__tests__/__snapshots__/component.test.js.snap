// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Banner of home page displaying 1`] = `
<div
  className="top-body"
>
  <div
    className="holder"
  >
    <p
      className="top-heading"
    >
      Gainers
    </p>
    <div
      className="icons"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 352 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M176 352c53.02 0 96-42.98 96-96V96c0-53.02-42.98-96-96-96S80 42.98 80 96v160c0 53.02 42.98 96 96 96zm160-160h-16c-8.84 0-16 7.16-16 16v48c0 74.8-64.49 134.82-140.79 127.38C96.71 376.89 48 317.11 48 250.3V208c0-8.84-7.16-16-16-16H16c-8.84 0-16 7.16-16 16v40.16c0 89.64 63.97 169.55 152 181.69V464H96c-8.84 0-16 7.16-16 16v16c0 8.84 7.16 16 16 16h160c8.84 0 16-7.16 16-16v-16c0-8.84-7.16-16-16-16h-56v-33.77C285.71 418.47 352 344.9 352 256v-48c0-8.84-7.16-16-16-16z"
        />
      </svg>
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M487.4 315.7l-42.6-24.6c4.3-23.2 4.3-47 0-70.2l42.6-24.6c4.9-2.8 7.1-8.6 5.5-14-11.1-35.6-30-67.8-54.7-94.6-3.8-4.1-10-5.1-14.8-2.3L380.8 110c-17.9-15.4-38.5-27.3-60.8-35.1V25.8c0-5.6-3.9-10.5-9.4-11.7-36.7-8.2-74.3-7.8-109.2 0-5.5 1.2-9.4 6.1-9.4 11.7V75c-22.2 7.9-42.8 19.8-60.8 35.1L88.7 85.5c-4.9-2.8-11-1.9-14.8 2.3-24.7 26.7-43.6 58.9-54.7 94.6-1.7 5.4.6 11.2 5.5 14L67.3 221c-4.3 23.2-4.3 47 0 70.2l-42.6 24.6c-4.9 2.8-7.1 8.6-5.5 14 11.1 35.6 30 67.8 54.7 94.6 3.8 4.1 10 5.1 14.8 2.3l42.6-24.6c17.9 15.4 38.5 27.3 60.8 35.1v49.2c0 5.6 3.9 10.5 9.4 11.7 36.7 8.2 74.3 7.8 109.2 0 5.5-1.2 9.4-6.1 9.4-11.7v-49.2c22.2-7.9 42.8-19.8 60.8-35.1l42.6 24.6c4.9 2.8 11 1.9 14.8-2.3 24.7-26.7 43.6-58.9 54.7-94.6 1.5-5.5-.7-11.3-5.6-14.1zM256 336c-44.1 0-80-35.9-80-80s35.9-80 80-80 80 35.9 80 80-35.9 80-80 80z"
        />
      </svg>
    </div>
  </div>
</div>
`;

exports[`Comapnies component renders correctly 1`] = `
Array [
  <div
    className="top-body"
  >
    <div
      className="holder"
    >
      <p
        className="top-heading"
      >
        Gainers
      </p>
      <div
        className="icons"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 352 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M176 352c53.02 0 96-42.98 96-96V96c0-53.02-42.98-96-96-96S80 42.98 80 96v160c0 53.02 42.98 96 96 96zm160-160h-16c-8.84 0-16 7.16-16 16v48c0 74.8-64.49 134.82-140.79 127.38C96.71 376.89 48 317.11 48 250.3V208c0-8.84-7.16-16-16-16H16c-8.84 0-16 7.16-16 16v40.16c0 89.64 63.97 169.55 152 181.69V464H96c-8.84 0-16 7.16-16 16v16c0 8.84 7.16 16 16 16h160c8.84 0 16-7.16 16-16v-16c0-8.84-7.16-16-16-16h-56v-33.77C285.71 418.47 352 344.9 352 256v-48c0-8.84-7.16-16-16-16z"
          />
        </svg>
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M487.4 315.7l-42.6-24.6c4.3-23.2 4.3-47 0-70.2l42.6-24.6c4.9-2.8 7.1-8.6 5.5-14-11.1-35.6-30-67.8-54.7-94.6-3.8-4.1-10-5.1-14.8-2.3L380.8 110c-17.9-15.4-38.5-27.3-60.8-35.1V25.8c0-5.6-3.9-10.5-9.4-11.7-36.7-8.2-74.3-7.8-109.2 0-5.5 1.2-9.4 6.1-9.4 11.7V75c-22.2 7.9-42.8 19.8-60.8 35.1L88.7 85.5c-4.9-2.8-11-1.9-14.8 2.3-24.7 26.7-43.6 58.9-54.7 94.6-1.7 5.4.6 11.2 5.5 14L67.3 221c-4.3 23.2-4.3 47 0 70.2l-42.6 24.6c-4.9 2.8-7.1 8.6-5.5 14 11.1 35.6 30 67.8 54.7 94.6 3.8 4.1 10 5.1 14.8 2.3l42.6-24.6c17.9 15.4 38.5 27.3 60.8 35.1v49.2c0 5.6 3.9 10.5 9.4 11.7 36.7 8.2 74.3 7.8 109.2 0 5.5-1.2 9.4-6.1 9.4-11.7v-49.2c22.2-7.9 42.8-19.8 60.8-35.1l42.6 24.6c4.9 2.8 11 1.9 14.8-2.3 24.7-26.7 43.6-58.9 54.7-94.6 1.5-5.5-.7-11.3-5.6-14.1zM256 336c-44.1 0-80-35.9-80-80s35.9-80 80-80 80 35.9 80 80-35.9 80-80 80z"
          />
        </svg>
      </div>
    </div>
  </div>,
  <img
    alt="Grow"
    className="cover-img"
    src="cover.jpg"
  />,
  <div>
    <div
      className="full-width"
    >
      <h5
        className="list-heading"
      >
        Top Performers
      </h5>
    </div>
    <div
      className="container"
    />
  </div>,
]
`;

exports[`Companies Data Passed 1`] = `
Array [
  <div
    className="full-width"
  >
    <h5
      className="list-heading"
    >
      Top Performers
    </h5>
  </div>,
  <div
    className="container"
  >
    <div
      className="container-child"
    >
      <div
        className="evenBox"
      >
        <svg
          className="arrow"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 448 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M190.5 66.9l22.2-22.2c9.4-9.4 24.6-9.4 33.9 0L441 239c9.4 9.4 9.4 24.6 0 33.9L246.6 467.3c-9.4 9.4-24.6 9.4-33.9 0l-22.2-22.2c-9.5-9.5-9.3-25 .4-34.3L311.4 296H24c-13.3 0-24-10.7-24-24v-32c0-13.3 10.7-24 24-24h287.4L190.9 101.2c-9.8-9.3-10-24.8-.4-34.3z"
          />
        </svg>
        <a
          aria-current={null}
          className="navLink"
          href="/companyGRVI"
          onClick={[Function]}
        >
          <h4
            className="companyName"
          >
            Grove, Inc.
          </h4>
        </a>
        <div
          className="combine-details"
        >
          <h3>
            GRVI
          </h3>
          <h3>
            $
            11.91
          </h3>
        </div>
      </div>
    </div>
    <div
      className="container-child"
    >
      <div
        className="oddBox"
      >
        <svg
          className="arrow"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 448 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M190.5 66.9l22.2-22.2c9.4-9.4 24.6-9.4 33.9 0L441 239c9.4 9.4 9.4 24.6 0 33.9L246.6 467.3c-9.4 9.4-24.6 9.4-33.9 0l-22.2-22.2c-9.5-9.5-9.3-25 .4-34.3L311.4 296H24c-13.3 0-24-10.7-24-24v-32c0-13.3 10.7-24 24-24h287.4L190.9 101.2c-9.8-9.3-10-24.8-.4-34.3z"
          />
        </svg>
        <a
          aria-current={null}
          className="navLink"
          href="/companyNRXP"
          onClick={[Function]}
        >
          <h4
            className="companyName"
          >
            NRx Pharmaceuticals, Inc.
          </h4>
        </a>
        <div
          className="combine-details"
        >
          <h3>
            NRXP
          </h3>
          <h3>
            $
            6.98
          </h3>
        </div>
      </div>
    </div>
  </div>,
]
`;

exports[`each company data shows 1`] = `
<div
  className="oddBox"
>
  <svg
    className="arrow"
    fill="currentColor"
    height="1em"
    stroke="currentColor"
    strokeWidth="0"
    style={
      Object {
        "color": undefined,
      }
    }
    viewBox="0 0 448 512"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M190.5 66.9l22.2-22.2c9.4-9.4 24.6-9.4 33.9 0L441 239c9.4 9.4 9.4 24.6 0 33.9L246.6 467.3c-9.4 9.4-24.6 9.4-33.9 0l-22.2-22.2c-9.5-9.5-9.3-25 .4-34.3L311.4 296H24c-13.3 0-24-10.7-24-24v-32c0-13.3 10.7-24 24-24h287.4L190.9 101.2c-9.8-9.3-10-24.8-.4-34.3z"
    />
  </svg>
  <a
    aria-current={null}
    className="navLink"
    href="/companyAAPL"
    onClick={[Function]}
  >
    <h4
      className="companyName"
    >
      APPLE INC
    </h4>
  </a>
  <div
    className="combine-details"
  >
    <h3>
      AAPL
    </h3>
    <h3>
      $
      450
    </h3>
  </div>
</div>
`;

exports[`renders correctly 1`] = `
Array [
  <div
    className="top-body"
  >
    <div
      className="holder"
    >
      <p
        className="top-heading"
      >
        Gainers
      </p>
      <div
        className="icons"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 352 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M176 352c53.02 0 96-42.98 96-96V96c0-53.02-42.98-96-96-96S80 42.98 80 96v160c0 53.02 42.98 96 96 96zm160-160h-16c-8.84 0-16 7.16-16 16v48c0 74.8-64.49 134.82-140.79 127.38C96.71 376.89 48 317.11 48 250.3V208c0-8.84-7.16-16-16-16H16c-8.84 0-16 7.16-16 16v40.16c0 89.64 63.97 169.55 152 181.69V464H96c-8.84 0-16 7.16-16 16v16c0 8.84 7.16 16 16 16h160c8.84 0 16-7.16 16-16v-16c0-8.84-7.16-16-16-16h-56v-33.77C285.71 418.47 352 344.9 352 256v-48c0-8.84-7.16-16-16-16z"
          />
        </svg>
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M487.4 315.7l-42.6-24.6c4.3-23.2 4.3-47 0-70.2l42.6-24.6c4.9-2.8 7.1-8.6 5.5-14-11.1-35.6-30-67.8-54.7-94.6-3.8-4.1-10-5.1-14.8-2.3L380.8 110c-17.9-15.4-38.5-27.3-60.8-35.1V25.8c0-5.6-3.9-10.5-9.4-11.7-36.7-8.2-74.3-7.8-109.2 0-5.5 1.2-9.4 6.1-9.4 11.7V75c-22.2 7.9-42.8 19.8-60.8 35.1L88.7 85.5c-4.9-2.8-11-1.9-14.8 2.3-24.7 26.7-43.6 58.9-54.7 94.6-1.7 5.4.6 11.2 5.5 14L67.3 221c-4.3 23.2-4.3 47 0 70.2l-42.6 24.6c-4.9 2.8-7.1 8.6-5.5 14 11.1 35.6 30 67.8 54.7 94.6 3.8 4.1 10 5.1 14.8 2.3l42.6-24.6c17.9 15.4 38.5 27.3 60.8 35.1v49.2c0 5.6 3.9 10.5 9.4 11.7 36.7 8.2 74.3 7.8 109.2 0 5.5-1.2 9.4-6.1 9.4-11.7v-49.2c22.2-7.9 42.8-19.8 60.8-35.1l42.6 24.6c4.9 2.8 11 1.9 14.8-2.3 24.7-26.7 43.6-58.9 54.7-94.6 1.5-5.5-.7-11.3-5.6-14.1zM256 336c-44.1 0-80-35.9-80-80s35.9-80 80-80 80 35.9 80 80-35.9 80-80 80z"
          />
        </svg>
      </div>
    </div>
  </div>,
  <img
    alt="Grow"
    className="cover-img"
    src="cover.jpg"
  />,
  <div>
    <div
      className="full-width"
    >
      <h5
        className="list-heading"
      >
        Top Performers
      </h5>
    </div>
    <div
      className="container"
    />
  </div>,
]
`;
